internal/docker/receiver/metrics_receiver.go

```diff
@@ -35,24 +37,24 @@ type resultV2 struct {
 	err       error
 }
 
-type metricsReceiver struct {
-	config   *Config
+type MetricsReceiver struct {
```

Comment:
Unsure how I feel about making all of these types and methods public. Is there an alternative to this? 

For example, I can create the OTLP exporter with fairly minimal exposure to the public API.

```go
t, err := otlphttpexporter.NewFactory().CreateTracesExporter(
		ctx,
		exporter.CreateSettings{
			ID:                component.NewID(component.MustNewType("mock_app_otlp_exporter")),
			TelemetrySettings: componenttest.NewNopTelemetrySettings(),
			BuildInfo:         component.NewDefaultBuildInfo(),
		},
		&otlphttpexporter.Config{
			ClientConfig: clientCfg,
			Encoding:     otlphttpexporter.EncodingProto,
		},
	)
```

Is there a way we could arrange this in a more effective way?

Reply:
I got an impression that this was necessary, but maybe this autoresolved at some point. Nevertheless, the PR is probably going to be split into smaller ones, so I will keep in mind your suggestion.